ion/textwriter_test.go

```diff
@@ -334,6 +334,69 @@ func TestWriteTextBadFinish(t *testing.T) {
 	}
 }
 
+func TestWriteTextPretty(t *testing.T) {
+	buf := strings.Builder{}
+	w := NewTextWriterOpts(&buf, TextWriterPretty)
+
+	w.BeginStruct()
+	{
+		w.FieldName("struct")
+		w.BeginStruct()
+		w.EndStruct()
+
+		w.FieldName("list")
+		w.Annotations("i", "am", "a", "list")
+		w.BeginList()
+		{
+			w.WriteString("value")
+			w.WriteNullType(StringType)
+		}
+		w.EndList()
+
+		w.FieldName("sexp")
+		w.BeginSexp()
+		{
+			w.WriteSymbol("+")
+			w.WriteInt(123)
+			w.BeginSexp()
+			{
+				w.WriteSymbol("*")
+				w.WriteInt(456)
+				w.WriteInt(789)
+			}
+			w.EndSexp()
+		}
+		w.EndSexp()
+	}
+	w.EndStruct()
+
+	if err := w.Finish(); err != nil {
+		t.Fatal(err)
+	}
+
+	actual := buf.String()
+	expected := `{
+	struct : {},
+	list : i :: am :: a :: list :: [
+		"value",
+		null.string
+	],
```

Comment:
List and Structs have `,` separators. Can we add to the test a nesting of these two Types (any combination is fine) and test that pretty print does what we expect for those as well, please.

Reply:
👍